workerpool/pool.go

```diff
@@ -140,7 +140,7 @@ func newJob(session *concurrency.Session, config *connection.Job, locksPrefix st
 func (j *job) start() {
 	for i := uint(0); i < j.numWorkers; i++ {
 		id := uint(j.id.JobNumber()*j.bucketSize) + i
-		worker := newWorker(id, j.connection, j.waitGroup, j.log.Named("worker"))
+		worker := newWorker(id, *j.connection, j.waitGroup, j.log.Named("worker"))
```

Comment:
Is this why you were asking earlier about each worker getting a copy of the Connection? I'm not sure I like the pointer deref here, would rather keep the pointer intact and just use that. In this case they're not consuming anything that could potentially clobber the other workers.